objects/src/notes/execution_hint.rs

```diff
@@ -10,15 +10,28 @@ use crate::NoteError;
 ///
 /// This struct can be represented as the combination of a tag, and a payload.
 /// The tag specifies the variant of the hint, and the payload encodes the hint data.
+///
+/// # Felt layout
+///
+/// [`NoteExecutionHint`] can be encoded into a [`Felt`] with the following layout:
+///
+/// ```text
+/// [26 zero bits | payload (32 bits) | tag (6 bits)]
+/// ```
```

Comment:
This is the expected layout that `kernel::tx::create_note` expects for the `execution_hint` argument. The only place where it is currently used is in the SWAP note construction:

https://github.com/0xPolygonMiden/miden-base/blob/7dfc1c3d96c5604a4d453c4ca72ba7a21759b94f/miden-lib/src/notes/mod.rs#L110-121

But it seems other user-created note types would also want to use it similarly.

Seems desirable to keep this as-is rather than make it two arguments, e.g. `execution_hint_tag` and `execution_hint_payload`.